### PR TITLE
finished fn-compareTimezones

### PR DESCRIPTION
**Pull request by mmvergara**
Finished compareTimezones function to compare users time

extra changes:
 - Promise.all to fetch both users at the same time. 
 - Extra validation

This is my first pr, writing a good pr desc is harder than js

### CODE REVIEW

File: src/db/timezones.ts [hunk 1]
```diff
@@ -12,18 +12,34 @@ async function addTimeZone(timezone: String, userID: String) {
   await userObject.save()
 }
 
-async function compareTimezones(user1: String, user2: String) {
-  // Find the users
-  const user1Object: any = await userModel.findOne({ id: user1 })
-  const user2Object: any = await userModel.findOne({ id: user2 })
+async function compareTimezones(user1Id: String, user2Id: String) {
+  // Find both users
+  const [user1Object, user2Object] = await Promise.all([
```
Comment (amelinium23) [inline src/db/timezones.ts:17]:
```suggestion
  const [firstUser, secondUser] = await Promise.all([
```

File: src/db/timezones.ts [hunk 2]
```diff
@@ -12,18 +12,34 @@ async function addTimeZone(timezone: String, userID: String) {
   await userObject.save()
 }
 
-async function compareTimezones(user1: String, user2: String) {
-  // Find the users
-  const user1Object: any = await userModel.findOne({ id: user1 })
-  const user2Object: any = await userModel.findOne({ id: user2 })
+async function compareTimezones(user1Id: String, user2Id: String) {
```
Comment (amelinium23) [inline src/db/timezones.ts:15]:
```suggestion
async function compareTimezones(firstUserId: String, secondUserId: String) {
```

File: src/db/timezones.ts [hunk 3]
```diff
@@ -12,18 +12,34 @@ async function addTimeZone(timezone: String, userID: String) {
...
+  const [user1Object, user2Object] = await Promise.all([
+    userModel.findOne({ id: user1Id }),
+    userModel.findOne({ id: user2Id }),
+  ])
 
   if (!user1Object || !user2Object) {
+    return "One or both of the users is not verified"
+  }
+
+  if (!user1Object?.timezone || !user2Object?.timezone) {
```
Comment (amelinium23) [inline src/db/timezones.ts:26]:
```suggestion
  if (!firstUser|| !secondUser) {
    return "One or both of the users is not verified"
  }

  if (!firstUser.timezone || !secondUser.timezone) {
```

File: src/db/timezones.ts [hunk 4]
```diff
@@ -12,18 +12,34 @@ async function addTimeZone(timezone: String, userID: String) {
...
-  // timezones are tored like this: UTC+1 or UTC-1 in the database
-  // I need to compare the timezones and return the difference
+  // Comparing Timezones
+  const user1 = new Date(
+    `Sun, 30 May 2021 14:59:15 ${user1Object.timezone}`
+  ).getTime()
+
+  const user2 = new Date(
+    `Sun, 30 May 2021 14:59:15 ${user2Object.timezone}`
+  ).getTime()
```
Comment (amelinium23) [inline src/db/timezones.ts:37]:
```suggestion
    `Sun, 30 May 2021 14:59:15 ${firstUser.timezone}`
  ).getTime()

  const user2 = new Date(
    `Sun, 30 May 2021 14:59:15 ${secondUser.timezone}`
  ).getTime()
```

File: src/db/timezones.ts [hunk 5]
```diff
@@ -12,18 +12,34 @@ async function addTimeZone(timezone: String, userID: String) {
...
+
+  const Status = user1 < user2 ? "ahead" : "behind"
+  const hours = Math.abs(user1 - user2) / 36e5
+
+  return `${user1Object.username} is ${hours}hrs ${Status} of ${user2Object.username}`
```
Comment (amelinium23) [inline src/db/timezones.ts:42]:
```suggestion
  return `${firstUser.username} is ${hours}hrs ${Status} of ${secondUser.username}`
```

**Review by amelinium23 — COMMENTED** [review]
Good changes small comments from me.

File: src/db/timezones.ts (hunk 2 above)
Comment (theyadev) [reply src/db/timezones.ts:15]:
```suggestion
async function compareTimezones(firstUserId: string, secondUserId: string) {
```

File: src/db/timezones.ts (hunk 4 above)
Comment (theyadev) [inline src/db/timezones.ts:37]:
```suggestion
  // Comparing Timezones
  const date = "Sun, 30 May 2021 14:59:15"
  
  const user1 = new Date(
    `${date} ${firstUser.timezone}`
  ).getTime()

  const user2 = new Date(
    `${date} ${secondUser.timezone}`
  ).getTime()
```

File: src/db/timezones.ts [hunk 6]
```diff
@@ -12,18 +12,34 @@ async function addTimeZone(timezone: String, userID: String) {
...
+  // Comparing Timezones
+  const user1 = new Date(
+    `Sun, 30 May 2021 14:59:15 ${user1Object.timezone}`
+  ).getTime()
+
+  const user2 = new Date(
+    `Sun, 30 May 2021 14:59:15 ${user2Object.timezone}`
+  ).getTime()
+
+  const Status = user1 < user2 ? "ahead" : "behind"
```
Comment (theyadev) [inline src/db/timezones.ts:39]: Why "Status" with an uppercase ?
```suggestion
  const status = user1 < user2 ? "ahead" : "behind"
```
You also have to modify the return statement

**Review by theyadev — CHANGES_REQUESTED** [review]

**Comment by mmvergara** [comment]
Fork Codebase was delete , will open a new PR.

**Pull request closed without merging**